RNA-seq/02-gastric_cancer_tximeta.Rmd

```diff
@@ -0,0 +1,174 @@
+---
+title: "Gastric cancer: gene-level summarization with `tximport`"
+output:   
+  html_notebook: 
+    toc: true
+    toc_float: true
+---
+
+**CCDL 2021**
+
+In this notebook, we'll import transcript level output from `salmon quant` and
+summarize it to the gene level using 'transcript expression import' - [`tximport`](https://bioconductor.org/packages/release/bioc/html/tximport.html) and 'tximport meta' - [`tximeta`](https://bioconductor.org/packages/release/bioc/html/tximeta.html). 
+
+
+![](diagrams/rna-seq_5.png)
+
+For more information about `tximeta`, see [this excellent vignette](https://www.bioconductor.org/packages/devel/bioc/vignettes/tximeta/inst/doc/tximeta.html) from Love, Soneson, Hickey, and Patro.
+
+We'll need the `quant.sf` files for all the samples in an experiment.
+
+## Libraries and functions
+
+```{r library, live = TRUE}
+# Load the tximeta package
+library(tximeta)
+
+# Load the SummarizedExperiment package
+library(SummarizedExperiment)
+```
+
+## Directories and files
+
+```{r directories, live = TRUE}
+# directory where the quant files are located, each sample is its own
+# directory
+quant_dir <- file.path("data", "gastric-cancer", "salmon_quant")
+
+# create a directory to hold the tximport results if it doesn't exist yet
+txi_dir <- file.path("data", "gastric-cancer", "tximport")
+if (!dir.exists(txi_dir)) {
+  dir.create(txi_dir, recursive = TRUE)
+}
+```
+
+```{r input-names}
+# the quant files themselves
+sf_files <- list.files(quant_dir, recursive = TRUE, full.names = TRUE,
+                       pattern = "quant.sf")
+```
+
+**Output**
+
+```{r output-names, live = TRUE}
+# Name the output gastric-cancer_tximport.RDS and use the directory created 
+# above as the rest of the path
+txi_out_file <- file.path(txi_dir, "gastric-cancer_tximport.RDS")
```

Comment:
Based on my interpretation @jashapiro's [comment here](https://github.com/AlexsLemonade/training-modules/pull/343#issuecomment-780807162)
> I think we can keep tximport in at least some of these places as long as we clearly lay out that the txi in tximeta stands for tximport (and the tx stands for transcript expression?). So the diagrams should be able to stay, and probably the directory names, but since the object structure is different we should probably change the .rds filenames

I believe we'll want to replace the `tximport` with `tximeta` here in the file name of the output RDS (please forgive me if I misinterpreted this comment!)

Reply:
I think you are right.